cmd/flux/bootstrap_git.go

```diff
@@ -192,7 +192,7 @@ func bootstrapGitCmdRun(cmd *cobra.Command, args []string) error {
 
 		// Configure repository URL to match auth config for sync.
 		repositoryURL.User = nil
-		repositoryURL.Scheme = "https"
+		repositoryURL.Scheme = repositoryURL.Scheme
```

Comment:
```suggestion
		if !bootstrapArgs.insecureHttpAllowed {
			repositoryURL.Scheme = "https"
		}
```

would probably be a better fix, to not blindly allow sending credentials over an insecure connection.